server/routes/changeGoal/ChangeGoalController.ts

```
@@ -34,9 +37,24 @@ export default class ChangeGoalController {
       const selectedAreaOfNeed = sortedAreasOfNeed.find(areaOfNeed => areaOfNeed.name === goal.areaOfNeed.name)
       const form = errors ? req.body : this.mapGoalToForm(goal)
 
+      const criminogenicNeedsData = req.services.sessionService.getCriminogenicNeeds()
       const planUuid = req.services.sessionService.getPlanUUID()
       const plan = await req.services.planService.getPlanByUuid(planUuid)
 
+      let assessmentDetailsForArea = null
+
+      // get assessment data
+      try {
+        const assessmentResponse = await req.services.assessmentService.getAssessmentByUuid(planUuid)
+        const assessmentData = assessmentResponse.sanAssessmentData
+
+        // 2. get the assessment details for the selected area of need
+        const areaConfig: AssessmentAreaConfig = areaConfigs.find(config => config.area === goal.areaOfNeed.name)
+        assessmentDetailsForArea = getAssessmentDetailsForArea(criminogenicNeedsData, areaConfig, assessmentData)
+      } catch {
+        /* swallow the error, missing data is handled in the template */
```

Same as a previous comment, can I ask the purpose and justification of an empty catch block here? I think it may be a little concerning to rely on the template to handle the error expressly as the template should primarily display the error outcome.

---

[Ditto.](https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/393#discussion_r1956077698)

---

Same response as noted [here](https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/393#discussion_r1956096869). Thanks for looking over it and explaining :-)